bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml

```diff
@@ -169,6 +169,14 @@
 		</channels>
 	</channel-group-type>
 
+	<channel-group-type id="oneCallMinutelyTimeSeries">
+		<label>One Call API Minutely Forecast Time Series</label>
```

Comment:
I'm wondering if you need to include "time series" in the label. States will be pushed to the event bus as they become current, so perhaps this should just be considered having a time dimension. I think this new channel group will eventually obsolete `oneCallMinutely`, since there is no longer a reason for keeping 60 channel groups like described here: https://www.openhab.org/addons/bindings/openweathermap/#one-call-api-minutely-forecast?

Reply:
Renamed in https://github.com/openhab/openhab-addons/pull/15963/commits/bd7350de0c7c352b03d956bd7e3e86d53c66d229.

In https://github.com/openhab/openhab-addons/pull/15963/commits/c8c73d47010572c8908f6073eaafb52c76120d71 I have updated the handler code to always fetch hourly, daily and minutely data from the API as this is needed for the time series channels and I changed the defaults of forecastHours and forecastDays to 0, so by default only the new time series channels are created when creating a new Thing.
So we slowly get rid of the large amount of channels without breaking anything.

Reply:
I'm wondering how that will actually work. I guess the defaults should also be provided here as well:
https://github.com/openhab/openhab-addons/blob/969518a077ba719d7fc5b02f461bd7b1331d91c6/bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/config/config.xml#L132-L146

But I guess that would be a breaking change for file-based configuration when these parameters are not explicitly set?

Reply:
The defaults are actually provided in https://github.com/openhab/openhab-addons/pull/15963/files#diff-586f2961d27e69af72e4b4e1637b6236b1c159eedbbe8c23b35c784a3ff66e95.

> But I guess that would be a breaking change for file-based configuration when these parameters are not explicitly set?

That is true, but IMHO this is no showstopper, since it is very easy to just explicitely set those in file-based configs.

Reply:
> The defaults are actually provided in

Sorry, I missed that.

> > But I guess that would be a breaking change for file-based configuration when these parameters are not explicitly set?

> That is true, but IMHO this is no showstopper, since it is very easy to just explicitely set those in file-based configs.

I agree it's not a showstopper, but it will cause immediate breakage after the upgrade requiring user attention for updating configuration to get it working again. That would probably also require a note in openhab-distro, because otherwise it could be easily missed until they realize that there's no data coming in anymore?

I'm thinking if this step is one release too soon? Perhaps we could keep the existing defaults, so users would have 4.0 to transition to time series channels?

Just some thoughts for consideration before we merge. If you think it's not a big problem, we can even merge as is.

Reply:
> That would probably also require a note in openhab-distro, because otherwise it could be easily missed until they realize that there's no data coming in anymore?

With that note and a note in the release notes, Insont see a real problem here, but …

> I'm thinking if this step is one release too soon? Perhaps we could keep the existing defaults, so users would have 4.0 to transition to time series channels?

I agree, so we can „bundle“ the default changes for the OneCall Thing with the changes for the other OWM Things and we have time to improve handling time series in rules.

Reply:
@jlaur I have just reverted changing the defaults a few seconds ago and updated the README to no note that the defaults will change in a future release and time series usage is encouraged.